### PR TITLE
Configure HTTP methods to capture in ASGI middleware and frameworks

### PR DESCRIPTION
**Pull request by antonpirker**
- Do not capture transactions for `OPTIONS` and `HEAD` HTTP methods by default. 
- Make it possible with an `http_methods_to_capture` config option for Starlette and FastAPI, to specify what HTTP methods to capture.

Fixes https://github.com/getsentry/sentry-python/issues/3529

Docs PR:
https://github.com/getsentry/sentry-docs/pull/11349

### CODE REVIEW

File: sentry_sdk/integrations/asgi.py [hunk 1]
```diff
@@ -185,52 +198,59 @@ async def _run_app(self, scope, receive, send, asgi_version):
...
-                                )
-                                if is_http_response:
-                                    transaction.set_http_status(event["status"])
+                                if transaction is not None:
+                                    is_http_response = (
+                                        event.get("type") == "http.response.start"
+                                        and "status" in event
+                                    )
+                                    if is_http_response:
+                                        transaction.set_http_status(event["status"])
```
Comment (antonpirker) [inline sentry_sdk/integrations/asgi.py:253]: I did it this way to make mypy happy

File: sentry_sdk/integrations/_asgi_common.py [hunk 2]
```diff
@@ -15,6 +15,19 @@
     from sentry_sdk.utils import AnnotatedValue
 
 
+DEFAULT_HTTP_METHODS_TO_CAPTURE = (
```
Comment (sentrivana) [inline sentry_sdk/integrations/_asgi_common.py:18]: Should we maybe just import this from `_wsgi_common` here? Or directly use the `_wsgi_common` constant in the integrations?

File: sentry_sdk/integrations/asgi.py [hunk 3]
```diff
@@ -55,6 +58,13 @@
 TRANSACTION_STYLE_VALUES = ("endpoint", "url")
 
 
+# This noop context manager can be replaced with "from contextlib import nullcontext" when we drop Python 3.6 support
+@contextmanager
+def nullcontext():
+    # type: () -> Iterator[None]
+    yield
```
Comment (sentrivana) [inline sentry_sdk/integrations/asgi.py:65]: Maybe move this to `utils.py` and use it from there? Also for `wsgi.py`

File: sentry_sdk/integrations/asgi.py (hunk 1 above)
Comment (sentrivana) [reply sentry_sdk/integrations/asgi.py:253]: mypy also made me do things I'm not proud of. We need mypy anonymous

**Review by sentrivana — APPROVED** [review]
See comments, we can dedup some things but looks very nice

**Comment by antonpirker** [comment]
I have merge this PR into https://github.com/getsentry/sentry-python/pull/3531 and did the de-duping in the other PR.

**Pull request merged**